connect/eaas/core/deployment/extension.py

```diff
@@ -58,19 +58,28 @@ def deploy_extension(repo, client, log, tag=None):
     environment = extension['environments'][arguments.get('env', 'production')]
     env_api = client('devops').services[extension['id']].environments[environment['id']]
 
-    environment = stop_environment(environment, env_api, log)
-    if environment['status'] == 'running':
-        return
-
     try:
-        log(f'Creating variables for extension with package_id {arguments["package_id"]}.')
-        process_variables(arguments, env_api)
+        log(f'Processing variables for extension with package_id {arguments["package_id"]}.')
+        vars_updated = process_variables(arguments, env_api, log)
+
+        if (
+            environment.get('git', {}).get('commit') != git['commit']
+            or environment['runtime'] == 'local'
+        ):
+            environment = stop_environment(environment, env_api, log)
+            if environment['status'] == 'running':
+                return
 
-        log(f'Updating and starting {environment["type"]} environment.')
-        env_api.update({'git': git, 'runtime': 'cloud'})
+            log(f'Updating {environment["type"]} environment.')
+            env_api.update({'git': git, 'runtime': 'cloud'})
```

Comment:
if this update transition the environment from local to cloud a deployment is started automatically. So you have to get the output of the update and, if the status is deploying -> job done.

Reply:
Yes, this scenario is taken into account. But I made another logic, further I call start of env only if runtime is 'cloud' and `environment['status'] == 'stopped'`

Reply:
👍